packages/vm/src/instance.rs

```diff
@@ -372,6 +374,26 @@ mod test {
         }
     }
 
+    #[test]
+    fn errors_in_imports_are_unwrapped_from_wasmer_errors() {
+        // set up an instance that will experience an error in an import
+        let mut instance = mock_instance_with_failing_api(&CONTRACT, &[]);
+        let init_result = call_init::<_, _, _, serde_json::Value>(
+            &mut instance,
+            &mock_env(&MockApi::new(MOCK_CONTRACT_ADDR.len()), "someone", &[]),
+            b"{\"verifier\": \"some1\", \"beneficiary\": \"some2\"}",
+        );
+
+        // in this case we get a `VmError::FfiError` rather than a `VmError::RuntimeErr` because the conversion
+        // from wasmer `RuntimeError` to `VmError` unwraps errors that happen in WASM imports.
+        match init_result.unwrap_err() {
+            VmError::FfiErr {
+                source: FfiError::Other { error, .. },
+            } if error == "canonical_address failed intentionally" => {}
```

Comment:
Nice check here and proving we do get the original error out. Cool stuff.

Reply:
I usually use
`assert_eq!(error, "canonical_address failed intentionally")`

instead of
`if error == "canonical_address failed intentionally" => {}`

But always curious to learn plus/minus of other approaches.

Reply:
well, both work, and i use both. In this case this just felt more natural